webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -583,6 +585,130 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
+      contents = editor.getValue();
+    }
+    if (context != null) {
+      _activeDataset = context;
+    } else {
+      _activeDataset = _openDataset.filter(dataset => dataset.active === true)[0];
+    }
+    const model = _activeDataset.model;
+    const fullName = _activeDataset.model.fileName;
+
+    _observable = new Observable((observer) => {
+      _observer = observer;
+    });
+
+    if (contents === undefined) {
+      this.snackBar.open(`Cannot retrieve contents to save`, "Close", {duration: MessageDuration.Medium, panelClass: 'center'});
+      return;
+    }
+
+    if (!destinationOverride && fullName) {
+      contents = editor.getValue().split('\n');
+      const requestUrl = ZoweZLUX.uriBroker.datasetContentsUri(fullName);
+      this.log.debug(`Should save contents to dataset. dataset=${fullName}, route=${requestUrl}`);
+      let result = this.isContentValidForDatasetWrite(contents, model.datasetAttrs);
+      if (result === true) {
+        this.saveDataset(context, _activeDataset, forceWrite, _observer, _observable);
+      } else {
+        this.snackBar.open(`Content invalid for saving to dataset. Reason=${result}`, 'Close', { duration:MessageDuration.Long, panelClass: 'center'});
+      }
+    } else {
+      //dataset is new or needs some alteration we can't do yet
+      this.snackBar.open(`${_activeDataset.name} could not be saved, feature not yet implemented.`, 
+                         'Close', { duration: MessageDuration.Long,   panelClass: 'center' });    
+    }
+    return _observable;
+  }
+
+  saveDataset(context: ProjectContext, activeDataset: ProjectContext, forceWrite: Boolean, _observer: Observer<void>, _observable: Observable<void>) {
+    const editor = this._editor.getValue();
+    let contents;
+    if(editor) {
+      contents = editor.getValue();
+    }
+    const model = activeDataset.model;
+    const fullName = model.fileName;
+    const etag = model.etag;
+    contents = editor.getValue().split('\n');
+    let headers = new Headers({'Etag': etag})
+    let reqBody = {records:contents};
+    if(etag) {
+      reqBody['etag'] = etag;
+    }
+    const requestUrl = ZoweZLUX.uriBroker.datasetContentsUri(fullName);
+    this.ngHttp.post(requestUrl, reqBody, {headers: headers, params:{force: forceWrite}}).subscribe(r => {
+      this.snackBar.open(`${activeDataset.name} has been saved!`,'Close', {duration:MessageDuration.Short, panelClass: 'center'});
+      /* Send buffer saved event */
+      this.bufferSaved.next({ buffer: activeDataset.model.contents, file: activeDataset.model.name });
+      this.openFileList.getValue()
+        .map(file => {
+          if (file.id === context.id) {
+            file.changed = false;
+          }
+          return file;
+        });
+      if (_observer != null) { _observer.next(null); }
+    }, e => {
+      const contentType = e.headers._headers.get('content-type');
+      if (contentType && contentType.indexOf('application/json; charset=utf-8') !== -1) {
```

Comment:
I don't really know what type of object 'e' is here. if contentType doesn't exist, it could also mean e.headers._headers doesn't exist, or maybe it does. You have an else case for if contentType doesn't exist, but this else case will never reach if e.headers is undefined or e.headers._headers is undefined. So could be good to check for those otherwise set contentType to null & continue